runtime/lua/vim/filetype.lua

```diff
@@ -863,6 +898,184 @@ local extension = {
       return "text"
     end
   end,
+  cmd = function (path, bufnr)
+    if getline(bufnr, 1):find("^/%*") then
+      return "rexx"
+    end
+    return "dosbatch"
+  end,
+  rul = function(path, bufnr)
+    if getline(bufnr, 1, 6):lower():find("installshield") then
+      return "ishd"
+    end
+    return "diva"
+  end,
+  cpy = function(path, bufnr)
+    if getline(bufnr, 1):find("^##") then
+      return "python"
+    end
+    return "cobol"
+  end,
+  dsl = function(path, bufnr)
+    if getline(bufnr, 1):find("^%s*<!") then
+      return "dsl"
+    end
+    return "structurizr"
+  end,
+  edf = "edif",
+  edfi = "edif",
+  edo = "edif",
+  edn = function(path, bufnr)
+    if getline(bufnr, 1):find("^%s*%(%s*edif>") then
+      return "edif"
+    end
+    return "clojure"
+  end,
+  smil = function(path, bufnr)
+    if getline(bufnr, 1):find("<?%s*xml.*?>") then
```

Comment:
`"<?%s*xml.*?>"` => `"^<?%s*xml.*?>$"` (since `=~` matches the whole string). The same change should be made anywhere else `=~` is used.

Reply:
`=~` does not match the whole string. However, this is still incorrect because the `?` characters need to be escaped:

```suggestion
    if getline(bufnr, 1):find("<%?%s*xml.*%?>") then
```